include/libtorrent/hasher512.hpp

```diff
@@ -114,7 +113,7 @@ namespace libtorrent
 #elif TORRENT_USE_COMMONCRYPTO
 		CC_SHA512_CTX m_context;
 #elif TORRENT_USE_CRYPTOAPI_SHA_512
-		HCRYPTHASH m_context;
+		aux::crypt_hash<CALG_SHA_512, PROV_RSA_AES> m_context;
```

Comment:
to pass arguments into the constructor, you do this: ``... m_context{CALC_SHA_512};``